cpp/src/interop/to_arrow_device.cu

```diff
@@ -164,27 +141,7 @@ int decimals_to_arrow(cudf::column_view input,
   nanoarrow::UniqueArray tmp;
   NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), NANOARROW_TYPE_DECIMAL128, input));
 
-  constexpr size_type BIT_WIDTH_RATIO = sizeof(__int128_t) / sizeof(DeviceType);
-  auto buf =
-    std::make_unique<rmm::device_uvector<DeviceType>>(input.size() * BIT_WIDTH_RATIO, stream, mr);
-
-  auto count = thrust::counting_iterator<size_type>(0);
-
-  thrust::for_each(
-    rmm::exec_policy(stream, mr),
-    count,
-    count + input.size(),
-    [in = input.begin<DeviceType>(), out = buf->data(), BIT_WIDTH_RATIO] __device__(auto in_idx) {
-      auto const out_idx = in_idx * BIT_WIDTH_RATIO;
-      // the lowest order bits are the value, the remainder
-      // simply matches the sign bit to satisfy the two's
-      // complement integer representation of negative numbers.
-      out[out_idx] = in[in_idx];
-#pragma unroll BIT_WIDTH_RATIO - 1
-      for (auto i = 1; i < BIT_WIDTH_RATIO; ++i) {
-        out[out_idx + i] = in[in_idx] < 0 ? -1 : 0;
-      }
-    });
+  auto buf = detail::decimals_to_arrow<DeviceType>(input, stream, mr);
```

Comment:
Could we declare and define the two `decimals_to_arrow` implementations together, and perhaps name them accordingly so it's clear why there's one implementation in to_arrow_host.cu and one in to_arrow_device.cu?

Reply:
I thought this PR was using the same implementation of `decimals_to_arrow` in both to_arrow_host.cu and to_arrow_device.cu and then using `populate_data_buffer` to cudaMemcpy() to host. Am I missing anything I am not aware of the design discussion but ideally `to_arrow_host.cu` should be entirely host impl.

Reply:
There's a function `detail::decimals_to_arrow` defined in `to_arrow_host` and this function calls that one. We'll want to refactor that call stack in your follow-up PR.